pkg/common/utils/resources.go

```diff
@@ -248,3 +255,32 @@ func (r *Resources) MultiplyBy(i int64) {
 	r.MemoryRequests.Set(r.MemoryRequests.Value() * i)
 	r.Disk.Set(r.Disk.Value() * i)
 }
+
+func (r *Resources) AddResources(resource Resources) {
+	r.CPULimits.Add(resource.CPULimits)
+	r.CPURequests.Add(resource.CPURequests)
+	r.MemoryLimits.Add(resource.MemoryLimits)
+	r.MemoryRequests.Add(resource.MemoryRequests)
+	r.Disk.Add(resource.Disk)
+}
+
+// AddPsqlSidecarResources adds the resource overhead for the PostgreSQL sidecar to the given resource.
+func (r *Resources) AddPsqlSidecarResources(instances int64) {
+	// Sidecar requests: 850m CPU, 1088Mi Memory
+	r.CPURequests.Add(*resource.NewMilliQuantity(850*instances, resource.DecimalSI))
+	r.MemoryRequests.Add(*resource.NewQuantity(1088*1024*1024*instances, resource.BinarySI))
```

Comment:
As discussed, please make these dynamically according to the definitions in the configMap.

Reply:
Refactored the code to get the values from the config